pandas/core/generic.py

```diff
@@ -3770,6 +3773,8 @@ def as_matrix(self, columns=None):
         --------
         pandas.DataFrame.values
         """
+        warnings.warn("This method will be removed in a future version. "
+                      "Use ``.values`` instead.")
```

Comment:
This needs to be a FutureWarning, and you also need to set a stacklevel (this is a top-level method, `stacklevel=2` should be the correct one)

Reply:
Can you also mention `as_matrix` explicitly in the message (that is clearer as you don't always directly see what line or what method on a certain line is causing the warning). 
And also follow a bit the typical message like "'as_matrix' is deprecated and will be removed in a future version. Use .."